src/Pages/RegisterForm/RegisterForm.tsx

```diff
@@ -107,56 +69,55 @@ const App: React.FC = () => {
     }));
   };
 
-  return (
-    <div className="form">
-      <div className="contact_us">
-        <span className="contact_us_text">Contact</span>
-        <span className="contact_us_text">us</span>
-      </div>
-      <Form
-        form={form}
-        name="form_item_path"
-        layout="vertical"
-        onFinish={onFinish}
-      >
-        <MyFormItemGroup prefix={['user']}>
-          <MyFormItemGroup prefix={['name']}>
-            <MyFormItem name="firstName" label="Name" required>
-              <Input
-                placeholder="Olivia Smith"
-                value={formData.firstName}
-                onChange={e => handleInputChange(e, 'firstName')}
-              />
-            </MyFormItem>
-            {errorName && <span className="input-warning">{errorName}</span>}
-            <MyFormItem name="businessEmail" label="Business email" required>
-              <Input
-                placeholder="example@business.com"
-                value={formData.businessEmail}
-                onChange={e => handleInputChange(e, 'businessEmail')}
-              />
-            </MyFormItem>
-            {errorEmail && <span className="input-warning">{errorEmail}</span>}
-          </MyFormItemGroup>
+  const handleModalClose = () => {
+    setIsSuccessful(null);
+    setIsModalVisible(false);
+  };
 
-          <MyFormItem name="companyName" label="Company Name" required>
-            <Input
-              placeholder="XRii"
-              value={formData.companyName}
-              onChange={e => handleInputChange(e, 'companyName')}
-            />
-          </MyFormItem>
-          {errorCompany && (
-            <span className="input-warning">{errorCompany}</span>
-          )}
-        </MyFormItemGroup>
+  const onFinish = async (values: any) => {
+    setLoading(true);
+    console.log(values);
 
-        <Button type="primary" block htmlType="submit">
-          Submit
-        </Button>
-      </Form>
+    await new Promise(resolve => setTimeout(resolve, 2000));
+    setIsSuccessful(Math.random() > 0.5);
+    setLoading(false);
+    setIsModalVisible(true);
+  };
+
+  return (
+    <div className="register_page">
+      <div className={`form ${isModalVisible ? 'blur-background' : ''}`}>
+        <div className="contact_us">
+          <span className="contact_us_text">Contact</span>
+          <span className="contact_us_text">us</span>
+        </div>
+        {isMobile ? (
+          <MobileForm
+            formData={formData}
+            handleInputChange={handleInputChange}
+            onFinish={onFinish}
+            formFields={formFields}
+            loading={loading}
+          />
+        ) : (
+          <DesktopForm
```

Comment:
i dont think this is a good idea to split the forms in 2 different components depending on a layout in your case. The functionality is the same, as far as i see that is the only difference. So that should be handled by some conditionals logic with the Form component.

it could be something like this 


`import` React, { useEffect, useState } from 'react';
import { Form } from 'antd';
import FormField from '../FormField/FormField';
import SubmitButton from '../SubmitButton/SubmitButton';
import './registerForm.scss';

const RegisterForm: React.FC = () => {
  const [isMobile, setIsMobile] = useState(window.innerWidth <= 768);
  const [formData, setFormData] = useState({ firstName: '', businessEmail: '', companyName: '' });
  const [loading, setLoading] = useState(false);

  const formFields = [
    { label: 'First Name', name: 'firstName', placeholder: 'Olivia Smith', rules: [{ required: true, message: 'Please input your name!' }] },
    { label: 'Business Email', name: 'businessEmail', placeholder: 'business@gmail.com', rules: [{ required: true, message: 'Please input your email!' }] },
    { label: 'Company Name', name: 'companyName', placeholder: 'Business Name', rules: [{ required: true, message: 'Please input your company name!' }] },
  ];

  useEffect(() => {
    const handleResize = () => setIsMobile(window.innerWidth <= 768);
    window.addEventListener('resize', handleResize);
    return () => window.removeEventListener('resize', handleResize);
  }, []);

  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>, field: string) => {
    setFormData(prevData => ({ ...prevData, [field]: e.target.value }));
  };

  const onFinish = async (values: any) => {
    setLoading(true);
    await new Promise(resolve => setTimeout(resolve, 2000));
    setLoading(false);
  };

  return (
    <Form
      className={`form-container ${isMobile ? 'form-mobile' : 'form-desktop'}`}
      name="register_form"
      layout={isMobile ? 'vertical' : 'horizontal'}
      onFinish={onFinish}
    >
      {formFields.map(field => (
        <FormField
          key={field.name}
          label={field.label}
          name={field.name}
          value={formData[field.name]}
          handleChange={handleInputChange}
          placeholder={field.placeholder}
          rules={field.rules}
        />
      ))}
      <Form.Item>
        <SubmitButton loading={loading} />
      </Form.Item>
    </Form>
  );
};

export default `RegisterForm;`

Reply:
Changed code and deleted desktop and mobile components https://github.com/olenakorol1708/my-react-app/tree/accordion/src/Pages/RegisterForm